fix: throw original error when ignore url is matched

fuzzybinary opened the pull request:
### What and why?

This should be in a patch release on 2.1.0. 

(cherry picked from commit 666a551268e5dc968d1846b7eb602ae8f7d1fbf4)

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

ronnnnn committed:
fix: throw original error when ignore url is matched

(cherry picked from commit 666a551268e5dc968d1846b7eb602ae8f7d1fbf4)

fuzzybinary committed:
Add changelog item for fix.

0xnm commented on packages/datadog_tracking_http_client/CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 
+## Unreleased
+
+* Fix _TypeError when request URL is matched ignoreUrlPatterns. See [#590] (Thanks [@ronnnnn][])
```
```suggestion
* Fix `_TypeError` when request URL is matched `ignoreUrlPatterns`. See [#590] (Thanks [@ronnnnn][])
```

fuzzybinary committed:
Pin iOS versions

This is now done on release but wasn't at the time of the release of this version.

fuzzybinary committed:
Update packages/datadog_tracking_http_client/CHANGELOG.md

Co-authored-by: Nikita Ogorodnikov <4046447+0xnm@users.noreply.github.com>

fuzzybinary committed:
Pin android as well.
